src/metadata/persistence/avu.clj

```diff
@@ -1,52 +1,92 @@
 (ns metadata.persistence.avu
-  (:use [korma.core :exclude [update]]
-        [korma.db :only [transaction]]
-        [slingshot.slingshot :only [throw+]])
-  (:require [kameleon.db :as db]
-            [korma.core :as sql]))
-
-(defn- target-where-clause
-  "Adds a where-clause to the given query for the given target."
-  [query target-type target-id]
-  (where query {:target_id   target-id
-                :target_type (db/->enum-val target-type)}))
+  (:use [slingshot.slingshot :only [throw+]])
+  (:require [metadata.util.db :refer [ds t]]
+            [next.jdbc :as jdbc]
+            [next.jdbc.plan :as plan]
+            [next.jdbc.sql :as jsql]
+            [next.jdbc.types :as jtypes]
+            [honey.sql :as sql]
+            [honey.sql.helpers :as h]
+            [kameleon.db :as db]))
+
+(def avu-columns [:id :attribute :value :unit :target_id :target_type :created_by :modified_by :created_on :modified_on])
+
+(defn- avus-base-query
+  [cols]
+  (-> (apply h/select cols)
+      (h/from (t "avus"))))
 
 (defn filter-targets-by-attrs-values
   "Finds the given targets that have any of the given attributes and values."
   [target-types target-ids attributes values]
-  (select :avus
-          (modifier "DISTINCT")
-          (fields :target_id
-                  :target_type)
-          (where {:target_id   [in target-ids]
-                  :target_type [in (map db/->enum-val target-types)]
-                  :attribute   [in attributes]
-                  :value       [in values]})))
+  (let [cols [:target_id :target_type]
+        q (-> (avus-base-query cols)
+              (dissoc :select) ;; change to select distinct
+              (h/select-distinct cols)
+              (h/where [:in :target_id target-ids]
+                       [:in :target_type (map jtypes/as-other target-types)]
```

Comment:
First common change: `jtypes/as-other` is how we make things with our enums work properly; we don't need our `->enum-val` any more.